src/System.CodeDom/tests/System/CodeDom/Compiler/CodeDomProviderTests.cs

```diff
@@ -134,6 +136,15 @@ public void CreateProvider_NoSuchLanguage_ThrowsConfigurationErrorsException(str
             AssertIsConfigurationErrorsException(ex2);
         }
 
+        [Fact]
+        public void CreateProvider_SerializeConfigurationErrorsException_ThrowsPlatformNotSupportedException()
+        {
+            Exception ex = Assert.ThrowsAny<Exception>(() => CodeDomProvider.CreateProvider(""));
```

Comment:
exception should be more specific (possibly will also fix @buyaa-n's comment)

Reply:
updated with a comment